fab/operations/offline.py

```diff
@@ -0,0 +1,54 @@
+import os
+
+from fabric.api import env, local
+from fab.utils import generate_bower_command
+from fab.const import (
+    OFFLINE_STAGING_DIR,
+    WHEELS_ZIP_NAME,
+    HQ_ZIP_NAME,
+    BOWER_ZIP_NAME,
+    NPM_ZIP_NAME,
+)
+
+
+def prepare_zipfiles():
+    local(
+        'git clone --depth 1 --recursive https://github.com/dimagi/commcare-hq.git {}/commcare-hq'
+        .format(OFFLINE_STAGING_DIR)
+    )
+
+    # After we've created our HQ code zip, let's get bower and npm
```

Comment:
I like that idea, but I'm not sure how easy that would be with submodules. https://github.com/dimagi/commcare-hq/blob/master/.gitmodules has the github url. and I'm not sure how easy it would be for that to sometimes be github.com and sometimes some private server. Could have some entry for github in /etc/hosts, but that doesn't seem like a good solution to me